gmail-to-salesforce-lead/main.bal

```diff
@@ -0,0 +1,193 @@
+        }
+    };
+}
+
+function getMatchingEmails(gmail:Client gmailClient, gmail:MailThread[] mailThreads) returns gmail:Message[] {
+    gmail:Message[] messages = [];
+    _ = from gmail:MailThread mailThread in mailThreads
+        do {
+            gmail:MailThread|error response = gmailClient->readThread(mailThread.id);
+            if response is error {
+                log:printError("An error occured while reading the email.", response, response.stackTrace(), threadId = mailThread.id);
+            } else {
+                messages.push((<gmail:Message[]>response.messages)[0]);
+            }
+        };
+
+    return messages;
+}
+
+function parseEmail(gmail:Message message) returns Email|error {
+    string 'from = <string>message.headerFrom;
+    string subject = <string>message.headerSubject;
+    string body = <string>(check mime:base64Decode(<string>(<gmail:MessageBodyPart>message.emailBodyInText).data));
+
+    return {
+        'from: 'from,
+        subject: subject,
+        body: body
+    };
+}
+
+function generateLead(string 'from, string subject, string body) returns Lead|error {
+    openAI:Client openAIClient = check new ({
+        auth: {token: openAIKey}
+    });
+
+    openAI:CreateChatCompletionRequest request = {
+        model: "gpt-3.5-turbo",
+        messages: [
+            {
+                role: "user",
+                content: string `
+            Extract the following details in JSON from the email.
+                {
+                    firstName__c: string, // Mandatory
+                    lastName__c: string, // Mandatory
+                    email__c: string // Mandatory
+                    phoneNumber__c: string, // With country code. Use N/A if unable to find
+                    company__c: string, // Mandatory
+                    designation__c: string // Not mandator. Use N/A if unable to find
+                }
+
+            Here is the email:    
+            {
+                from: ${'from},
+                subject: ${subject},
+                body: ${body}
+            }
+        `
+            }
+        ]
+    };
+
+    openAI:CreateChatCompletionResponse response = check openAIClient->/chat/completions.post(request);
+
+    Lead result = check (<string>response.choices[0].message?.content).fromJsonStringWithType(Lead);
+    return result;
+}
+
+function addLeadsToSalesforce(Lead[] leads) returns error? {
+    sfdc:Client sfdcClient = check new ({
+        baseUrl: salesforceBaseUrl,
+        auth: {token: salesforceAccessToken}
+    });
+
+    from Lead lead in leads
+    do {
+        sfdc:CreationResponse|error createResponse = check sfdcClient->create("EmailLead__c", lead);
+        if createResponse is error {
+            log:printError("An error occured while creating a Lead object on salesforce.", createResponse, createResponse.stackTrace(), lead = lead);
```

Comment:
Break at max char length of 120 https://learn-ballerina.github.io/best_practices/wrap_code.html

Reply:
Fixed in [cce2eaf](https://github.com/ballerina-guides/integration-samples/pull/32/commits/cce2eafff5580b9fc4de69c431a8951e6c90b36a)